src/extension/index.ts

```diff
@@ -0,0 +1,205 @@
+import { loadLogs } from './loadLogs';
+import { Panel } from './panel';
+import { regionToSelection } from './regionToSelection';
+import { ResultDiagnostic } from './resultDiagnostic';
+import { Store } from './store';
+import { update, updateChannelConfigSection } from './update';
+
+export async function activate(context: ExtensionContext) {
+    const disposables = context.subscriptions;
+    Store.extensionPath = context.extensionPath;
+    Store.globalState = context.globalState;
+    disposables.push(commands.registerCommand('sarif.clearState', () => {
+        context.globalState.update('view', undefined);
+        commands.executeCommand('workbench.action.reloadWindow');
+    }));
+    const store = new Store();
+
+    // Basing
+    const urisNonSarif = await workspace.findFiles('**/*', '.sarif'); // Ignore folders?
+    const fileAndUris = urisNonSarif.map(uri => [uri.path.split('/').pop(), uri.path]) as [string, string][];
+    const baser = new Baser(mapDistinct(fileAndUris), store);
+
+    // Panel
+    const panel = new Panel(context, baser, store);
+    disposables.push(commands.registerCommand('sarif.showPanel', () => panel.show()));
+
+    // General Activation
+    activateDiagnostics(store, baser);
+    activateWatchDocuments(store, panel);
+    activateDecorations(store, panel);
+    activateVirtualDocuments(store);
+
+    // Check for Updates
+    // Borrowed from: https://github.com/Microsoft/vscode-languageserver-node/blob/db0f0f8c06b89923f96a8a5aebc8a4b5bb3018ad/client/src/main.ts#L217
+    const isDebugOrTestMode =
+        process.execArgv.some(arg => /^--extensionTestsPath=?/.test(arg)) // Debug
+        && process.execArgv.some(arg => /^--(debug|debug-brk|inspect|inspect-brk)=?/.test(arg)); // Test
+    if (!isDebugOrTestMode) {
+        disposables.push(workspace.onDidChangeConfiguration(event => {
+            if (!event.affectsConfiguration(updateChannelConfigSection)) return;
+            update();
+        }));
+        update();
+    }
+
+    // API
+    return {
+        async openLogs(logs: Uri[], _options: unknown, cancellationToken?: CancellationToken) {
+            store.logs.push(...await loadLogs(logs, cancellationToken));
+            if (cancellationToken?.isCancellationRequested) return;
+            if (store.results.length) panel.show();
+        },
+        async closeLogs(logs: Uri[]) {
+            for (const uri of logs) {
+                store.logs.removeWhere(log => log._uri === uri.toString());
+            }
+        },
+        async closeAllLogs() {
+            store.logs.splice(0);
+        },
+        get uriBases() {
+            return baser.uriBases.map(path => Uri.file(path)) as ReadonlyArray<Uri>;
+        },
+        set uriBases(values) {
+            baser.uriBases = values.map(uri => uri.path);
+        },
+    };
+}
+
+function activateDiagnostics(store: Store, baser: Baser) {
+    const diagsAll = languages.createDiagnosticCollection('SARIF');
```

Comment:
>diagsAll [](start = 10, length = 8)

You should return this value from the function and make sure it is added to the disposables array as it is disposable.